lib/services/protocol_service.ts

```diff
@@ -612,7 +611,9 @@ export class SNProtocolService extends PureService implements EncryptionDelegate
       );
     });
     let decryptedPayloads;
-    if (keyParams) {
+    let keyParams;
+    if (keyParams = data.keyParams || data.auth_params) {
```

Comment:
Hmm..don't really like the style of this. It may work but it's not a pattern we use anywhere else in the lib.

What about

```javascript
const keyParamsData = data.keyParams || data.auth_params;
if(keyParamsData) {
   const keyParams = this.createKeyParams(keyParamsData);
}
```